drivers/sensor/apds9253/apds9253.c

```diff
@@ -121,9 +121,91 @@ static int apds9253_channel_get(const struct device *dev, enum sensor_channel ch
 	return 0;
 }
 
+static int apds9253_attr_set_gain(const struct device *dev, uint8_t gain)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t value;
+
+	if (drv_data->gain == gain) {
+		return 0;
+	}
+
+	static uint8_t value_map[] = {
+		APDS9253_LS_GAIN_RANGE_1, APDS9253_LS_GAIN_RANGE_3,  APDS9253_LS_GAIN_RANGE_6,
+		APDS9253_LS_GAIN_RANGE_9, APDS9253_LS_GAIN_RANGE_18,
+	};
+
+	if (gain < APDS9253_LS_GAIN_RANGE_1 || gain > APDS9253_LS_GAIN_RANGE_18) {
+		return -EINVAL;
+	}
+	value = value_map[gain];
+
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_GAIN_REG, APDS9253_LS_GAIN_MASK,
+				   (value & APDS9253_LS_GAIN_MASK))) {
+		LOG_ERR("Not able to set light, sensor gain is not set");
+		return -EIO;
+	}
+
+	drv_data->gain = gain;
+
+	return 0;
+}
+
+static int apds9253_attr_set_meas_rate_mes(const struct device *dev, uint8_t meas_rate_mes)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t value;
+
+	if (drv_data->meas_rate_mes == meas_rate_mes) {
+		return 0;
+	}
+
+	static uint8_t value_map[] = {
+		APDS9253_LS_MEAS_RATE_MES_2000MS, APDS9253_LS_MEAS_RATE_MES_1000MS,
+		APDS9253_LS_MEAS_RATE_MES_500MS,  APDS9253_LS_MEAS_RATE_MES_200MS,
+		APDS9253_LS_MEAS_RATE_MES_100MS,  APDS9253_LS_MEAS_RATE_MES_50MS,
+		APDS9253_LS_MEAS_RATE_MES_25MS};
+
+	if (meas_rate_mes < APDS9253_LS_MEAS_RATE_MES_25MS ||
+	    meas_rate_mes > APDS9253_LS_MEAS_RATE_MES_2000MS) {
+		return -EINVAL;
+	}
+	value = value_map[meas_rate_mes];
+
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_MEAS_RATE_REG,
+				   APDS9253_LS_MEAS_RATE_MES_MASK,
+				   (value & APDS9253_LS_MEAS_RATE_MES_MASK))) {
+		LOG_ERR("Not able to set light sensor measurement rate is not set");
+		return -EIO;
+	}
+
+	drv_data->meas_rate_mes = meas_rate_mes;
+
+	return 0;
+}
+
+static int apds9253_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	switch (attr) {
+	case SENSOR_ATTR_GAIN:
+		return apds9253_attr_set_gain(dev, val->val1);
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return apds9253_attr_set_meas_rate_mes(dev, val->val1);
```

Comment:
The units here are in Hz as per the sensor.h spec. So you'll need to convert them to ms and find the closest match from your sensor's options.